orangecontrib/text/widgets/owguardian.py

```diff
@@ -0,0 +1,203 @@
+from datetime import datetime, timedelta, date
+
+from AnyQt.QtCore import Qt
+from AnyQt.QtWidgets import QApplication, QFormLayout
+
+from Orange.data import StringVariable
+from Orange.widgets.settings import Setting
+from Orange.widgets.widget import OWWidget, Msg
+from Orange.widgets.credentials import CredentialManager
+from Orange.widgets import gui
+
+from orangecontrib.text.corpus import Corpus
+from orangecontrib.text.guardian import TheGuardianCredentials, TheGuardianAPI
+from orangecontrib.text.widgets.utils import CheckListLayout, QueryBox, DatePickerInterval, gui_require, asynchronous
+
+
+class IO:
+    CORPUS = 'Corpus'
+
+
+class OWGuardian(OWWidget):
+    class CredentialsDialog(OWWidget):
+        name = 'The Guardian Credentials'
+        want_main_area = False
+        resizing_enabled = False
+        cm_key = CredentialManager('The Guardian API Key')
+        key_input = ''
+
+        class Error(OWWidget.Error):
+            invalid_credentials = Msg('These credentials are invalid.')
+
+        def __init__(self, parent):
+            super().__init__()
+            self.parent = parent
+            self.api = None
+
+            form = QFormLayout()
+            form.setContentsMargins(5, 5, 5, 5)
+            self.key_edit = gui.lineEdit(self, self, 'key_input', controlWidth=350)
```

Comment:
Could we have `self.key_edit.setText('test')`. This key [seems to work](https://content.guardianapis.com/search?q=debates&api-key=test) for testing purposes ...